src/fastcs/transport/epics/ioc.py

```diff
@@ -243,23 +239,13 @@ def datatype_updater(datatype: DataType):
 def _create_and_link_write_pv(
     pv_prefix: str, pv_name: str, attr_name: str, attribute: AttrW[T]
 ) -> None:
-    if attr_is_enum(attribute):
-
-        async def on_update(value):
-            await attribute.process_without_display_update(
-                enum_index_to_value(attribute, value)
-            )
-
-        async def async_write_display(value: T):
-            record.set(enum_value_to_index(attribute, value), process=False)
+    cast_method = get_cast_method_from_epics_type(attribute.datatype)
 
-    else:
+    async def on_update(value):
+        await attribute.process_without_display_update(cast_method(value))
 
-        async def on_update(value):
-            await attribute.process_without_display_update(value)
-
-        async def async_write_display(value: T):
-            record.set(value, process=False)
+    async def async_write_display(value: T):
+        record.set(cast_method(value), process=False)
```

Comment:
This should be casting _to_ an EPICS type, but this is using the same method as `on_update`.